av1an-core/src/project.rs

```diff
@@ -387,6 +387,18 @@ impl Project {
       self.video_params = self.encoder.get_default_arguments();
     }
 
+    if self.encoder == Encoder::aom
+      && self
+        .video_params
+        .iter()
+        .any(|param| param == "--enable-keyframe-filtering=2")
+      && self.concat != ConcatMethod::MKVMerge
```

Comment:
one more nitpick, `&&` evaluates left to right, so the check for mkvmerge should be before the param check for enable-kf=2 just to avoid comparing strings if it's not necessary